tests/unit/routes/requests-test.js

```diff
@@ -0,0 +1,102 @@
+var nock = require('nock')
+var test = require('tap').test
+
+var getServer = require('../utils/get-server')
+// var couchdbErrorTests = require('../utils/couchdb-error-tests')
+
+getServer(function (error, server) {
+  if (error) {
+    return test.fail(error)
+  }
+
+  test('POST /requests password reset', function (group) {
+    var postRequestRouteOptions = {
+      method: 'POST',
+      url: '/requests',
+      payload: {
+        type: 'createresetpassword',
+        username: 'john'
+      }
+    }
+    function postRequestResponseMock () {
+      return nock('http://localhost:5984')
+        // mock valid response to sign in request
+        .get('/_users/org.couchdb.user%3Ajohn')
+        .reply(200, {
+          _id: 'org.couchdb.user:john',
+          _rev: '1-abc',
+          name: 'john',
+          type: 'user',
+          email: 'john@email.com',
+          roles: []
+        })
+        .get('/uuids')
+        .reply(200, {
+          uuids: ['9aef3ffae9597ce60dfec0fee200094c']
+        })
+        .put('/_users/org.couchdb.user%3Ajohn', function (body) {
+          return body.reset_token === '9aef3ffae9597ce60dfec0fee200094c'
+        })
+        .reply(200)
+    }
+
+    group.test('Creates token and timestamp for username', function (t) {
```

Comment:
Very good point with the timestamp. For handling timeouts, I guess the simplest thing would be to set timers in the request handlers, plus an initial bootstrap on startup?